### ISSUE TITLE
MySQL `rank()` causes an invalid frame clause in re-written SQL

### ISSUE THREAD

**Issue by peasee** [spiceai/spiceai#3076]
Partitioning without a frame clause introduces an unsupported frame clause in MySQL

```
rank() over (
  partition by grouping(i_category)+grouping(i_class),
  case when grouping(i_class) = 0 then i_category end
  order by sum(ws_net_paid) desc)
```

is rewritten with a frame clause like `NULLS FIRST RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW` which is unsupported in MySQL.

Additionally, MySQL ignores frame clauses for `rank()`, so we should exclude it altogether in the rewritten SQL: https://dev.mysql.com/doc/refman/8.4/en/window-functions-frames.html

Affects TPC-DS Q86, Q70, Q67, Q57, Q51, Q49, Q47, Q44 and Q36

**Comment by sgrebnov** [comment]
Should be fixed by https://github.com/spiceai/datafusion/pull/42

**Comment by peasee** [comment]
> Should be fixed by [spiceai/datafusion#42](https://github.com/spiceai/datafusion/pull/42)

~~Looks like this has caused a regression with the AST analyzer. I'll be updating the analyzer to fix the regression.~~ This isn't a regression. I was just testing a query that didn't have `ORDER BY` inside a `sum()` window function.
